CedarJavaFFI/src/objects.rs

```diff
@@ -122,8 +111,21 @@ impl<'a> JEntityTypeName<'a> {
 impl<'a> Object<'a> for JEntityTypeName<'a> {
     fn cast(env: &mut JNIEnv<'a>, obj: JObject<'a>) -> Result<Self> {
         assert_is_class(env, &obj, "com/cedarpolicy/value/EntityTypeName")?;
-
-        Ok(Self { obj })
+        let namespace = env.call_method(&obj, "getNamespace", "()Ljava/util/List;", &[])?;
+        let namespace_components = List::cast_unchecked(get_object_ref(namespace)?, env)?;
+        let basename = env.call_method(&obj, "getBaseName", "()Ljava/lang/String;", &[])?;
+        let basename = JString::cast(env, get_object_ref(basename)?)?;
+        let jstr_basename = env.get_string(&basename)?;
+        let basename_str = String::from(jstr_basename);
+        let mut full_type_name: Vec<String> = jstr_list_to_rust_vec(env, &namespace_components)?;
+        full_type_name.push(basename_str);
+        let has_namespace_component_with_colon = full_type_name.iter().any(|s| s.contains("::"));
+        if has_namespace_component_with_colon {
+            return Err("components of the type name cannot contain colons".into());
+        }
+        let full_ns_str: String = full_type_name.join("::");
+        let type_name: EntityTypeName = full_ns_str.parse()?;
```

Comment:
This looks like the same code as L42-51. Might be nice to centralize this logic in case we end up changing handling in the future.